arkouda/dataframe.py

```diff
@@ -104,14 +106,48 @@ class DataFrameGroupBy:
         If True the grouped values of the aggregation keys will be treated as an index.
     """
 
-    def __init__(self, gb, df, gb_key_names=None, as_index=True):
+    def __init__(self, gb, df, gb_key_names=None, as_index=True, dropna=True):
+
         self.gb = gb
         self.df = df
         self.gb_key_names = gb_key_names
         self.as_index = as_index
         for attr in ["nkeys", "permutation", "unique_keys", "segments"]:
             setattr(self, attr, getattr(gb, attr))
 
+        self.dropna = dropna
+        self.where_not_nan = None
+        self.all_non_nan = False
+
+        if dropna:
+            from arkouda import all as ak_all
+            from arkouda import isnan
+
+            # calculate ~isnan on each key then & them all together
+            # keep up with if they're all_non_nan, so we can skip indexing later
+            key_cols = (
+                [df[k] for k in gb_key_names] if isinstance(gb_key_names, List) else [df[gb_key_names]]
+            )
+            where_key_not_nan = [
```

Comment:
`where_key_not_nan` is already computed in `groupbyclass.py` at line 272.  I guess it's one of those time memory trade offs whether it's better to recompute or save it the first time.

https://github.com/Bears-R-Us/arkouda/blob/master/arkouda/groupbyclass.py#L272

Reply:
yaaah I saw this and it did annoy me. but I couldn't think of a great way to take advantage of this without having the list of arrays persist indefinitely (by making it a field of groupby). But I also didn't try too hard to think of a way. The recalculation shouldn't be too expensive tho